include/zim/item.h

```diff
@@ -23,28 +23,26 @@
 
 #include "zim.h"
 #include "blob.h"
+#include "entry.h"
 #include <string>
 
 namespace zim
 {
-  class Dirent;
-  class FileImpl;
-
   /**
    * An `Item` in an `Archive`
    *
    * All `Item`'s methods are threadsafe.
    */
-  class Item
+  class Item : private Entry
   {
     public: // types
       typedef std::pair<std::string, offset_type> DirectAccessInfo;
 
     public: // functions
-      explicit Item(std::shared_ptr<FileImpl> file_, entry_index_type idx_);
+      explicit Item(const Entry& entry);
```

Comment:
Do we need this constructor public ?

The previous constructor was public but it was impossible to use as `FileImpl` was not publicly defined and it was impossible to get a pointer to it.
But now one could create an `Item` from an `Entry`. There is a assert in the constructor implementation but it would be better to make the constructor private.

Reply:
Done